write-fonts/src/graph.rs

```diff
@@ -203,17 +203,62 @@ impl Graph {
         }
     }
 
+    /// Attempt to pack the graph.
+    ///
+    /// This involves finding an order for objects such that all offsets are
+    /// resolveable.
+    ///
+    /// In the simple case, this just means finding a topological ordering.
+    /// In exceptional cases, however, this may require us to significantly
+    /// modify the graph.
+    ///
     /// returns `true` if a solution is found, `false` otherwise
-    pub(crate) fn topological_sort(&mut self) -> bool {
+    pub(crate) fn pack_objects(&mut self) -> bool {
+        if self.basic_sort() {
+            return true;
+        }
+
+        self.assign_32bit_spaces();
+        self.sort_shortest_distance();
+
+        let n_overflows = self.find_overflows().len();
+        if n_overflows > 0 {
+            log::info!("packing failed with {n_overflows} overflows");
+        }
+        n_overflows == 0
+    }
+
+    /// Initial sorting operation. Attempt kahn, falling back to shortest distance.
```

Comment:
Kahn. Also link to https://github.com/harfbuzz/harfbuzz/blob/main/docs/repacker.md#topological-sorting-algorithms? And perhaps the HarfBuzz implementation of the equivalent?